### PR TITLE
Bug 1224136 - Reset icon size on window resize. r=mhenretty

### PR DESCRIPTION
**Pull request by Cwiiis**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8f4c7a93cf11f3ea16eb9791cf2e2ca972d45824

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b4a3df44bf93e592de1f59ad9abff430e5c84e08

File: apps/homescreen/js/app.js [hunk 1]
```diff
@@ -122,7 +122,11 @@
       this.exitEditMode();
     });
 
+    // Icon and grid sizing behaviour
     this._iconSize = 0;
+    this.lastWindowWidth = window.innerWidth;
+    this.lastWindowHeight = window.innerHeight;
+    this.windowResizeTimeout = null;
```
Comment (mikehenrty) [inline apps/homescreen/js/app.js]: What does this do?

File: shared/elements/gaia-site-icon/script.js [hunk 2]
```diff
@@ -334,7 +346,8 @@ window.GaiaAppIcon = (function(exports) {
       if (!this._hasUserSetIcon) {
         // Process the image
         var canvas = document.createElement('canvas');
-        canvas.width = canvas.height = this._size;
+        var size = this.size * window.devicePixelRatio;
```
Comment (mikehenrty) [inline shared/elements/gaia-site-icon/script.js]: Seems like we fetch this adjusted size multiple times. Might be worth having a getter or function for it.

File: apps/homescreen/js/app.js (hunk 1 above)
Comment (Cwiiis) [inline apps/homescreen/js/app.js]: Oh, was a remnant of something else I was trying, will remove.

File: shared/elements/gaia-site-icon/script.js (hunk 2 above)
Comment (Cwiiis) [inline shared/elements/gaia-site-icon/script.js]: That's a good idea... Will have to fix in follow-up, this is an external component.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e4730b5d9acb776e11f40646118b79ef0c0b47fc

**Pull request merged**